core/session.ts

```diff
@@ -367,8 +404,13 @@ export class Session {
 
   private checkActionNameIsUnused(name: string) {
     // Check for duplicate names
-    if (this.tables[name] || this.operations[name] || this.assertions[name]) {
-      const message = `Duplicate action name detected. Names must be unique across tables, assertions, and operations: "${name}"`;
+    if (
+      this.tables[name] ||
+      this.operations[name] ||
+      this.assertions[name] ||
+      this.declarations[name]
```

Comment:
Good to know! If it goes in this list then it is my understanding that declarations will also contain a schema in front of the name (just as tables, ops and assertions). If this change is approved before mine then I'll need to adjust my code accordingly. Do let me know what do you think makes more sense to merge earlier, either this one or mine.

Reply:
Yup, that's correct, declarations *must* have a schema too.